examples/src/main/java/org/apache/flink/agents/examples/agents/EmbeddingsAgent.java

```diff
@@ -0,0 +1,378 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.agents.examples.agents;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.flink.agents.api.Agent;
+import org.apache.flink.agents.api.InputEvent;
+import org.apache.flink.agents.api.OutputEvent;
+import org.apache.flink.agents.api.annotation.Action;
+import org.apache.flink.agents.api.annotation.EmbeddingModelConnection;
+import org.apache.flink.agents.api.annotation.EmbeddingModelSetup;
+import org.apache.flink.agents.api.annotation.Tool;
+import org.apache.flink.agents.api.annotation.ToolParam;
+import org.apache.flink.agents.api.context.RunnerContext;
+import org.apache.flink.agents.api.resource.ResourceDescriptor;
+import org.apache.flink.agents.integrations.embeddingmodels.ollama.OllamaEmbeddingModelConnection;
+import org.apache.flink.agents.integrations.embeddingmodels.ollama.OllamaEmbeddingModelSetup;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An agent that generates embeddings for each row of data using Ollama embedding models.
+ *
+ * <p>This agent receives text data, processes it to generate high-dimensional vector embeddings,
+ * and outputs the results with metadata. It demonstrates how to integrate embedding models into
+ * Flink Agents workflows for vector-based processing and similarity search applications.
+ *
+ * <p>The agent supports various embedding models available in Ollama such as: - nomic-embed-text
+ * (768 dimensions) - mxbai-embed-large (1024 dimensions) - all-minilm (384 dimensions)
+ */
+public class EmbeddingsAgent extends Agent {
```

Comment:
I think `EmbeddingAgent` and `WorkflowEmbeddingsAgentExampleJob` are more like e2e test for verification purposes, rather than user-facing examples. We probably don't want to recommend users to directly call the embedding models in their actions. Therefore, I'd suggest to move these to the `e2e-test` module.